crates/pep508-rs/src/marker.rs

```diff
@@ -967,6 +967,87 @@ impl MarkerTree {
+            // Ex) `extra == 'dev'`
+            if expression.l_value == MarkerValue::Extra {
+                if let MarkerValue::QuotedString(r_string) = &expression.r_value {
+                    if let Ok(r_extra) = ExtraName::from_str(r_string) {
+                        return extras.contains(&r_extra);
+                    }
+                }
+            }
+            // Ex) `'dev' == extra`
+            if expression.r_value == MarkerValue::Extra {
+                if let MarkerValue::QuotedString(l_string) = &expression.l_value {
+                    if let Ok(l_extra) = ExtraName::from_str(l_string) {
+                        return extras.contains(&l_extra);
+                    }
+                }
+            }
+            false
+        }
+
+        match self {
+            Self::Expression(expression) => {
+                // If the expression is true, we can remove the marker entirely.
+                if is_true(&expression, extras) {
+                    None
+                } else {
+                    // If not, return the original marker.
+                    Some(Self::Expression(expression))
+                }
+            }
+            Self::And(expressions) => {
+                // Remove any expressions that are _true_ due to the presence of an extra.
+                let simplified = expressions
+                    .into_iter()
+                    .filter_map(|marker| marker.simplify_extras(extras))
+                    .collect::<Vec<_>>();
+
+                // If there are no expressions left, return None.
+                if simplified.is_empty() {
+                    None
+                } else if simplified.len() == 1 {
+                    // If there is only one expression left, return the simplified marker.
+                    simplified.into_iter().next()
+                } else {
+                    // If there are still expressions left, return the simplified marker.
+                    Some(Self::And(simplified))
+                }
+            }
+            Self::Or(expressions) => {
+                let num_expressions = expressions.len();
+
+                // Remove any expressions that are _true_ due to the presence of an extra.
+                let simplified = expressions
+                    .into_iter()
+                    .filter_map(|marker| marker.simplify_extras(extras))
+                    .collect::<Vec<_>>();
+
+                // If any of the expressions are true, the entire marker is true.
+                if simplified.len() < num_expressions {
+                    None
+                } else if simplified.len() == 1 {
+                    // If there is only one expression left, return the simplified marker.
+                    simplified.into_iter().next()
+                } else {
+                    // If there are still expressions left, return the simplified marker.
+                    Some(Self::Or(simplified))
```

Comment:
It's a little confusing that the outcome in these comments is the same?

Reply:
The previous branch could be "return the remaining expression".